languageserver/integration/codelenses.go

```diff
@@ -69,75 +72,41 @@ func (i *FlowIntegration) codeLenses(
 func (i *FlowIntegration) showDeployContractAction(
 	uri protocol.DocumentUri,
 	program *ast.Program,
-) *protocol.CodeLens {
-
-	// Do not show deploy button when no active account exists
-	if i.activeAddress == flow.EmptyAddress {
-		return nil
-	}
+	version float64,
+	checker *sema.Checker,
+) ([]*protocol.CodeLens, error) {
+	i.updateContractInfoIfNeeded(uri, version, checker)
 
-	contract := program.SoleContractDeclaration()
+	contractInfo := i.contractInfo[uri]
+	kind := contractInfo.kind
 
-	if contract == nil {
-		return nil
+	if kind == contractTypeUnknown || contractInfo.startPos == nil {
+		return nil, nil
 	}
 
-	name := contract.Identifier.Identifier
+	name := contractInfo.name
+	position := *contractInfo.startPos
+	signersList := contractInfo.pragmaSignersStrings[:]
 
-	position := contract.StartPosition()
+	codelensRange := conversion.ASTToProtocolRange(position, position)
+	var codeLenses []*protocol.CodeLens
 
-	return &protocol.CodeLens{
-		Range: conversion.ASTToProtocolRange(
-			position,
-			position,
-		),
-		Command: &protocol.Command{
-			Title: fmt.Sprintf(
-				"deploy to account 0x%s",
-				i.activeAddress.Hex(),
-			),
-			Command:   CommandDeployContract,
-			Arguments: []interface{}{uri, name},
-		},
+	// Check emulator state
+	emulatorStateLens := i.checkEmulatorState(codelensRange)
+	if emulatorStateLens != nil {
+		return []*protocol.CodeLens{emulatorStateLens}, nil
 	}
-}
 
-// showDeployContractInterfaceAction shows a deploy button when there is exactly one contract interface declaration,
-// and no other actionable declarations
-//
-func (i *FlowIntegration) showDeployContractInterfaceAction(
-	uri protocol.DocumentUri,
-	program *ast.Program,
-) *protocol.CodeLens {
-
-	// Do not show deploy button when no active account exists
-	if i.activeAddress == flow.EmptyAddress {
-		return nil
+	// TODO: resolve check for amount of signers equals to provided by pragma
```

Comment:
Is this still a TODO for this PR or a future one?

Reply:
Solved it in this one ^__^